bot/bot.py

```diff
@@ -0,0 +1,50 @@
+import logging
+import os
+
+import disnake
+from disnake.ext import commands
+
+logger = logging.getLogger(__name__)
+
+try:
+    import dotenv
+    dotenv.load_dotenv()
+except ModuleNotFoundError:
+    pass
+
+DEV_LOG_CHANNEL = os.environ.get("DEV_LOG_CHANNEL")
+TOKEN = os.environ.get("TOKEN")
+
+
+class SirRobin(commands.Bot):
+    """Sir-Robin core."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.name = "Sir-Robin"
+        self.dev_log = int(DEV_LOG_CHANNEL)
+
+    def add_cog(self, cog: commands.Cog) -> None:
+        """Delegate to super to register `cog`.
+
+        This only serves to make the info log, so that extensions don't have to.
+        """
+        super().add_cog(cog)
+        logger.info(f"Cog loaded: {cog.qualified_name}")
+
+
+bot = SirRobin(command_prefix="&", DEV_LOG=DEV_LOG_CHANNEL)
+
+
+@bot.event
+async def on_ready():
+    logger.info("'on_ready' event hit")
+    devlog = bot.get_channel(bot.dev_log)  # noqa: F841
+    icon = bot.user.display_avatar.url
+
+    embed = disnake.Embed(title="Sir Robin", description="Sir Robin online")
+    embed.set_author(name=bot.name, icon_url=icon)
+    await devlog.send(embed=embed)
```

Comment:
On top of the comment saying that this should be done in the subclass, this doesn't need to be in the `on_ready` event, and can be done with `Bot.wait_until_guild_available()` method (see [sir-lancebot's implementation](https://github.com/python-discord/sir-lancebot/blob/main/bot/bot.py#L126)).